lib/progressrus.rb

```diff
@@ -2,29 +2,60 @@
 require 'securerandom'
 require 'redis'
 require 'time'
-require_relative "progressrus/store"
 require_relative "progressrus/store/base"
 require_relative "progressrus/store/redis"
 require_relative "progressrus/store/progressbar"
 require_relative "progressrus/core_ext/enumerable"
 
 class Progressrus
+  class InvalidStoreError < StandardError
+    def initialize
+      message = <<~MSG
+        The store needs to implement `persist`, `scope`, `find` and `flush`
+        We have a base class that your store can inherit from:
+          Progressrus::Store::Base
+      MSG
+      super(message)
+    end
+  end
+
   class << self
+    def clear_stores
+      @stores = {}
+    end
+
     def stores
```

Comment:
I was going to say that we might want to use a mutex so this memoization doesn't become thread unsafe for someone who uses this gem with Puma or Sidekiq - but this is completely outside of the scope if your changes.